### ISSUE TITLE
ESLint 4.0 alpha: False positive for space-before-function-paren

### ISSUE THREAD

**Issue by Trott** [eslint/eslint#8430]
**Tell us about your environment**

* **ESLint Version:** v4.0.0-alpha.0
* **Node Version:** v7.8.0
* **npm Version:** 4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
env:
  node: true
  es6: true

parserOptions:
  ecmaVersion: 2017

rules:
  space-before-function-paren: [2, never]
```

**What did you do? Please include the actual source code causing the issue.**

```js
'use strict';
const assert = require('assert');
const util = require('util');

assert.strictEqual(util.inspect(async () => {}), '[AsyncFunction]');
```

**What did you expect to happen?**
I expected no errors to be reported because the docs say that async arrow functions are ignored by default.

**What actually happened? Please include the actual, raw output from ESLint.**

Got this instead:

```
  5:38  error  Unexpected space before function parentheses  space-before-function-paren
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. This was actually an [intentional change](https://github.com/eslint/eslint/commit/7dd890db194faec55cfd0485118c1a8fdf0b076d), but I forgot to update the docs.
